### PR TITLE
Whereabouts should implement the reconciliation controller

### PR DESCRIPTION
**Pull request by dougbtv**
This adds yaml to start a controller in replacement of the cronjob

### CODE REVIEW

File: bindata/network/multus/multus.yaml [hunk 1]
```diff
@@ -521,50 +522,68 @@ spec:
...
-apiVersion: batch/v1
-kind: CronJob
+apiVersion: apps/v1
+kind: DaemonSet
 metadata:
-  name: ip-reconciler
+  name: whereabouts-reconciler
   namespace: openshift-multus
   labels:
     tier: node
```
Comment (andreaskaris) [inline bindata/network/multus/multus.yaml:533]: Question: is there a reason to have that label here?  It seems to be the only thing using that label:
~~~
[akaris@linux cluster-network-operator (whereabouts-reconciler-controller-yaml-fix)]$ oc get ds -A -l tier=node
NAMESPACE          NAME                     DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR              AGE
openshift-multus   whereabouts-reconciler   6         6         6       6            6           kubernetes.io/arch=amd64   21m
[akaris@linux cluster-network-operator (whereabouts-reconciler-controller-yaml-fix)]$ oc get deployment -A -l tier=node
No resources found
[akaris@linux cluster-network-operator (whereabouts-reconciler-controller-yaml-fix)]$ oc get pods -A -l tier=node
NAMESPACE          NAME                           READY   STATUS    RESTARTS   AGE
openshift-multus   whereabouts-reconciler-2nghd   1/1     Running   0          21m
openshift-multus   whereabouts-reconciler-5smsm   1/1     Running   0          21m
openshift-multus   whereabouts-reconciler-cww4v   1/1     Running   0          21m
openshift-multus   whereabouts-reconciler-lfm8d   1/1     Running   0          21m
openshift-multus   whereabouts-reconciler-xd8nz   1/1     Running   0          21m
openshift-multus   whereabouts-reconciler-xnh67   1/1     Running   0          21m
~~~

File: bindata/network/multus/multus.yaml [hunk 2]
```diff
@@ -521,50 +522,68 @@ spec:
...
+              fieldPath: metadata.namespace
+        resources:
+          requests:
+            cpu: "100m"
+            memory: "100Mi"
+          limits:
+            cpu: "100m"
+            memory: "200Mi"
+        securityContext:
+          privileged: true
```
Comment (andreaskaris) [inline bindata/network/multus/multus.yaml:576]: Can we narrow this down?

**Review by andreaskaris — CHANGES_REQUESTED** [review]
2 questions and I fixed up the unit tests here: https://github.com/dougbtv/cluster-network-operator/pull/1
The rest looks good to me but I might have to dig into the whereabouts code further to fully grasp what the reconciler is doing

File: bindata/network/multus/multus.yaml [hunk 3]
```diff
@@ -521,50 +522,68 @@ spec:
...
-  jobTemplate:
+  selector:
+    matchLabels:
+      name: whereabouts-reconciler
+  updateStrategy:
+    type: RollingUpdate
+  template:
+    metadata:
+      labels:
+        tier: node
```
Comment (nicklesimba) [inline bindata/network/multus/multus.yaml:544]: Echoing @andreaskaris comment regarding label.

File: bindata/network/multus/multus.yaml [hunk 4]
```diff
@@ -436,6 +436,7 @@ spec:
               "kubernetes": {
                 "kubeconfig": "/etc/kubernetes/cni/net.d/whereabouts.d/whereabouts.kubeconfig"
               },
+              "reconciler_cron_expression": "30 4 * * *",
```
Comment (nicklesimba) [inline bindata/network/multus/multus.yaml:439]: This looks like we're running reconciler once a day at 4:30am - is this what we want?

**Review by nicklesimba — COMMENTED** [review]
Looking good. I have a couple of small questions/nits, otherwise solid work.

File: bindata/network/multus/multus.yaml (hunk 1 above)
Comment (dougbtv) [reply bindata/network/multus/multus.yaml:533]: good catch! removing.

File: bindata/network/multus/multus.yaml (hunk 2 above)
Comment (dougbtv) [reply bindata/network/multus/multus.yaml:576]: Good idea. I divided it in half, up for recommendations, too.

File: bindata/network/multus/multus.yaml (hunk 4 above)
Comment (dougbtv) [reply bindata/network/multus/multus.yaml:439]: Thanks for the check -- yeah, I think once a day for the like "audit style reconciliation" is good. Happy to take other input, too

**Comment by dougbtv** [comment]
/retitle Whereabouts should implement the reconciliation controller

File: bindata/network/multus/multus.yaml (hunk 2 above)
Comment (andreaskaris) [reply bindata/network/multus/multus.yaml:572]: haha I actually meant the securityContext.privileged ;-)

File: bindata/network/multus/multus.yaml (hunk 3 above)
Comment (andreaskaris) [reply bindata/network/multus/multus.yaml:540]: So removing the label is still outstanding, because it's still here on the latest commit :-)

**Review by andreaskaris — COMMENTED** [review]
1 nit (label) and one thing to thinkg about (securityContext) outstanding.
I think you can amend the commit message and remove the signed-off-by: Andreas Karis and replace it with yourself, given that I'm not the commit author

**Comment by andreaskaris** [comment]
/lgtm
Let's fix up the 2 nits in another commit given that Doug is OOO this week

**Pull request merged**